src/core/qgsmaplayer.cpp

```diff
@@ -1166,7 +1167,7 @@ bool QgsMapLayer::importNamedStyle( QDomDocument &myDocument, QString &myErrorMe
   }
 #endif
 
-  return readSymbology( myRoot, myErrorMessage );
+  return readSymbology( myRoot, myErrorMessage, QgsPathResolver() ); // TODO: support relative paths?
```

Comment:
@wonder-sk are my comments ^ not in-line with what this work could be extended to do, i.e. should project-relative symbol paths be implemented in another way/place?